aodndata/soop/soop_tmv_nrt.py

```diff
@@ -0,0 +1,154 @@
+import os
+import re
+from datetime import datetime
+
+import numpy as np
+import pandas as pd
+from aodncore.pipeline import HandlerBase, PipelineFilePublishType, PipelineFile
+from aodncore.util.misc import get_pattern_subgroups_from_string
+from ncwriter import DatasetTemplate
+from netCDF4 import date2num, Dataset
+from pkg_resources import resource_filename
+
+from ship_callsign import ship_callsign_list
+
+SHIP_CODE = 'VLST'
+
+SOOP_NRT_LOG_PATTERN = re.compile(r"""
+                                  EPA_
+                                  (?P<facility>SOOP_TMV1)_
+                                  (?P<product_code>D2M|M2D|MEL|DEV|M2S|S2M|SYD)_.*
+                                  \.log$
+                                  """, re.VERBOSE)
+
+NC_JSON_TEMPLATE_MOORING = resource_filename("aodndata", "templates/soop_tmv_nrt_nc_template_mooring.json")
+NC_JSON_TEMPLATE_TRAJECTORY = resource_filename("aodndata", "templates/soop_tmv_nrt_nc_template_trajectory.json")
+
+
+def parse_log_file(log_path):
+    df = pd.read_csv(log_path, header=None,
+                     engine='python')
+    df.columns = ["TIME", "unknown1", "product_code", "unknown2", "LATITUDE", "LONGITUDE", "TEMP", "PSAL",
+                  "CPHL", "TURB"]
+    date_format = '%Y/%m/%d %H:%M:%S'
+    df['TIME'] = pd.to_datetime(df['TIME'], format=date_format)
+
+    df = df[np.isfinite(df['LATITUDE'])]
+    df = df[np.isfinite(df['LONGITUDE'])]
+
+    # force make time monotonic and time not duplicated
+    df.set_index('TIME', inplace=True)
+    df.sort_index(axis=0, inplace=True)
+    df = df[~df.index.duplicated(keep='first')]
+    return df
+
+
+def netcdf_writer(log_path, output_dir):
+    df = parse_log_file(log_path)
+    ship_callsign_ls = ship_callsign_list()
+    log_filename = os.path.basename(log_path)
+
+    if SHIP_CODE not in ship_callsign_ls:
+        raise ValueError(
+            "Missing vessel callsign in file name '{name}'.".format(name=log_filename))
+
+    if SOOP_NRT_LOG_PATTERN.match(log_filename):
+        fields = get_pattern_subgroups_from_string(log_filename, SOOP_NRT_LOG_PATTERN)
+        product_code = fields['product_code']
+
+    if product_code in ['D2M', 'M2D']:
```

Comment:
thats fine because it's in the regex above
```python
SOOP_NRT_LOG_PATTERN = re.compile(r"""
                                  EPA_
                                  (?P<facility>SOOP_TMV1)_
                                  (?P<product_code>D2M|M2D|MEL|DEV|M2S|S2M|SYD)_.*
                                  \.log$
                                  """, re.VERBOSE)
```